data/metric/source_type.go

```diff
@@ -48,6 +54,11 @@ func (t *SourceType) String() string {
 	return ""
 }
 
+// String fulfills stringer interface, returning empty string on invalid source types.
+func (t *SourceType) IsPositive() bool {
```

Comment:
@varas I used a pointer receiver here because there was already a method with pointer receiver for the struct, I read some place that mixing pointer and value receivers for the same struct is a bad practice, the only ugly thing is that I had to reference the pointer to make the comparison.

Do you have any suggestion for this? or should I leave it like this?